tests/test_servers/test_servlet.py

```diff
@@ -75,3 +79,39 @@ def test_get_obj_does_not_exist(self, test_servlet):
         assert resp.output_type == "exception"
         error = deserialize_data(resp.data["error"], "pickle")
         assert isinstance(error, KeyError)
+
+    @pytest.mark.level("local")
+    def test_failed_cluster_servlet(self):
+
+        # need to initialize ray so the cluster servlet will be initialized when calling get_cluster_servlet.
+        ray.init(
+            ignore_reinit_error=True,
+            logging_level=logging.ERROR,
+            namespace="runhouse",
+        )
+
+        invalid_cluster_config = {
+            "api_server_url": "https://api.run.house.invalid",
+            "name": "mocked-cluster",
+        }
+
+        cluster_servlet = get_cluster_servlet(
+            create_if_not_exists=True,
+            cluster_config=invalid_cluster_config,
+            name="invalid_cluster_servlet",
+        )
+        try:
+            # waiting for a seconds so the async cluster status check thread will be executed and then fail and kill the clsuter servlet.
+            time.sleep(1)
+            cluster_servlet = ray.get_actor(
+                name="invalid_cluster_servlet", namespace="runhouse"
+            )
+            # if the cluster servlet is not deleted -> it is not none -> not cluster_servlet = False ->
+            # the test fails, because we expect that the cluster servlet will be deleted
+            assert not cluster_servlet
```

Comment:
is this supposed to return an empty object, or raise a ValueError?

Reply:
It suppose to raise an ValueError, with the msg we are asserting when we catch the exception (`"Failed to look up actor with name 'invalid_cluster_servlet'. This could because 1. You are trying to look up a named actor you didn't create. 2. The named actor died. 3. You did not use a namespace matching the namespace of the actor."`). If such exception is raised => the assertion of the msg is `True` and the test passes.

If the ValueError is not raised => ray found the `invalid_cluster_servlet` (it is not deleted) => `cluster_servlet != None` => `assert not cluster_servlet` is `false` => the test fails.

Reply:
what would be the reason why this test behaves differently on each run? like why wouldn't an exception always be raised? 

if we do expect the test to raise the error, we should do: 
```
with pytest.raises(ValueError):
   ...
```